File: CPP/Set/Minimum Number of Operations to Make Elements in Array Distinct.cpp
class Solution {
public:
    bool check (vector<int>&nums, int index){
        set<int>st;
        for (int i = index; i < nums.size(); i++) {
        if (st.count(nums[i])) {
            return false; // found duplicate
        }
        st.insert(nums[i]);
    }
    return true; // all unique
    }
    int minimumOperations(vector<int>& nums) {
        int ans=0;
        for(int i=0;i<nums.size();i+=3){
            if(check(nums,i))break;
            ans++;
        }
        return ans;
    }
};

File: Python/Set/Minimum Number of Operations to Make Elements in Array Distinct.py
class Solution:
    def check(self,nums:List[int], index:int)->bool:
        st=set()
        for i in range(index,len(nums)):
            if nums[i] in st:
                return False
            st.add(nums[i])
        return True


    def minimumOperations(self, nums: List[int]) -> int:
        ans=0
        for i in range(0,len(nums),3):
            if self.check(nums,i):
                break
            ans+=1
        return ans
